contentstore/models.py

```diff
@@ -27,3 +36,79 @@ def __unicode__(self):
             rfield(self.minute), rfield(self.hour), rfield(self.day_of_week),
             rfield(self.day_of_month), rfield(self.month_of_year),
         )
+
+
+class MessageSet(models.Model):
+
+    """
+        Details about a set of messages that a recipient can be sent on
+        a particular schedule
+    """
+    short_name = models.CharField(_('Short name'), max_length=20, unique=True)
+    notes = models.TextField(_('Notes'), null=True, blank=True)
+    next_set = models.ForeignKey('self',
+                                 null=True,
+                                 blank=True)
+    default_schedule = models.ForeignKey(Schedule,
+                                         related_name='message_sets',
+                                         null=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __unicode__(self):
+        return "%s" % self.short_name
+
+
+def generate_new_filename(instance, filename):
+    ext = filename.split('.')[-1]  # get file extension
+    return "%s.%s" % (datetime.now().strftime("%Y%m%d%H%M%S%f"), ext)
+
+
+class BinaryContent(models.Model):
+    """
+        File store for reference in messages. Storage method handle by
+        settings file.
+    """
+
+    content = models.FileField(upload_to=generate_new_filename,
+                               max_length=100)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __unicode__(self):
+        return "%s" % (self.content.path.split('/')[-1])
+
+
+class Message(models.Model):
+
+    """
+        Messages that a recipient can be sent
+    """
+    messageset = models.ForeignKey(MessageSet,
+                                   related_name='messages',
+                                   null=False)
+    sequence_number = models.IntegerField(null=False, blank=False)
+    lang = models.CharField(max_length=6, null=False, blank=False)
+    text_content = models.TextField(null=True, blank=True)
+    binary_content = models.ForeignKey(BinaryContent,
+                                       related_name='message',
+                                       null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ['sequence_number']
+
+    def clean(self):
+        # Don't allow messages to have neither a text or binary content
+        if self.text_content is None and self.binary_content is None:
+            raise ValidationError(
+                _('Messages must have text or file attached'))
+
+    def save(self, *args, **kwargs):
+        self.clean()
+        super(Message, self).save(*args, **kwargs)
```

Comment:
It's "called" not returned. See: https://docs.djangoproject.com/en/1.8/topics/db/models/#overriding-model-methods